dipy/viz/ui.py

```diff
@@ -2161,3 +2163,185 @@ def handle_move_callback(self, i_ren, obj, slider):
+            The corresponding image .
+        """
+        imgExt = imgPath.split(".")[-1].lower()
+        if imgExt == "png":
+            png = vtk.vtkPNGReader()
+            png.SetFileName(imgPath)
+            png.Update()
+            img = png.GetOutput()
+        elif imgExt in ["jpg", "jpeg"]:
+            jpeg = vtk.vtkJPEGReader()
+            jpeg.SetFileName(imgPath)
+            jpeg.Update()
+            img = jpeg.GetOutput()
+        else:
+            error_msg = "This file format is not supported by the Image Holder"
+            warn(Warning(error_msg))
+        return img
+
+    def _get_size(self):
+        lower_left_corner = self.texture_points.GetPoint(0)
+        upper_right_corner = self.texture_points.GetPoint(2)
+        size = np.array(upper_right_corner) - np.array(lower_left_corner)
+        return abs(size[:2])
+
+    def _setup(self):
+        """ Setup this UI Component.
+        Return an image as a 2D actor with a specific position.
+
+        Returns
+        -------
+        :class:`vtkTexturedActor2D`
+        """
+        self.texture_polydata = vtk.vtkPolyData()
+        self.texture_points = vtk.vtkPoints()
+        self.texture_points.SetNumberOfPoints(4)
+
+        polys = vtk.vtkCellArray()
+        polys.InsertNextCell(4)
+        polys.InsertCellPoint(0)
+        polys.InsertCellPoint(1)
+        polys.InsertCellPoint(2)
+        polys.InsertCellPoint(3)
+        self.texture_polydata.SetPolys(polys)
+
+        tc = vtk.vtkFloatArray()
+        tc.SetNumberOfComponents(2)
+        tc.SetNumberOfTuples(4)
+        tc.InsertComponent(0, 0, 0.0)
+        tc.InsertComponent(0, 1, 0.0)
+        tc.InsertComponent(1, 0, 1.0)
+        tc.InsertComponent(1, 1, 0.0)
+        tc.InsertComponent(2, 0, 1.0)
+        tc.InsertComponent(2, 1, 1.0)
+        tc.InsertComponent(3, 0, 0.0)
+        tc.InsertComponent(3, 1, 1.0)
+        self.texture_polydata.GetPointData().SetTCoords(tc)
+
+        texture_mapper = vtk.vtkPolyDataMapper2D()
+        if major_version <= 5:
+            texture_mapper.SetInput(self.texture_polydata)
+        else:
+            texture_mapper.SetInputData(self.texture_polydata)
+
+        image = vtk.vtkTexturedActor2D()
+        image.SetMapper(texture_mapper)
+
+        self.texture = vtk.vtkTexture()
+        image.SetTexture(self.texture)
+
+        image_property = vtk.vtkProperty2D()
+        image_property.SetOpacity(1.0)
+        image.SetProperty(image_property)
+        self.actor = image
+
+        # Add default events listener to the VTK actor.
+        self.handle_events(self.actor)
+
+    def _get_actors(self):
+        """ Returns the actors that compose this UI component.
+        """
+        return [self.actor]
+
+    def _add_to_renderer(self, ren):
+        """ Add all subcomponents or VTK props that compose this UI component.
+
+        Parameters
+        ----------
+        ren : renderer
+        """
+        ren.add(self.actor)
+
+    def resize(self, size):
```

Comment:
I think this could be refactored to just use size as a property. Let me know what you think @karandeepSJ

Reply:
Size is not an attribute in the UI class or in any other classes. It would be best to maintain that uniformity and keep it as `resize()`

Reply:
You're correct. The way I thought of this is that size is an important attribute for some elements. 
For instance, in the TextBlock/TextBox, text is a major attribute and therefore we make that a property. Why not do the same for size? 
Maybe we can keep it the way it is for now and later refactor all elements if it makes sense.